custom_components/ttlock/coordinator.py

```diff
@@ -139,14 +145,26 @@ async def _async_update_data(self) -> LockState:
                 try:
                     state = await self.api.get_lock_state(self.lock_id)
                     new_data.locked = state.locked == State.locked
+                    new_data.opened = state.opened == SensorState.opened
                 except Exception:
                     pass
 
             new_data.auto_lock_seconds = details.autoLockTime
+            new_data.auto_lock = True if details.autoLockTime > 0 else False
+            new_data.lock_sound = details.lockSound == 1
+
             new_data.passage_mode_config = await self.api.get_lock_passage_mode_config(
                 self.lock_id
             )
-
+            
+            if Features.door_sensor in Features.from_feature_value(details.featureValue):
```

Comment:
>    Although I added the featureValue, I'm having troubles using this as -
>    Just because the lock has sensor capabilities that doesn't mean there is actually a sensor.
>    Not sure how only make 1 querySensor call if there isnt a sensor present

This might be something to roll in with the timestamp to prevent polling to frequently.
If we can detect somehow that no sensor is installed, just set it to something that means we don't poll for it again. The user would have to reload the integration after installing the sensor, but that seems ok to me given that we don't want to burn the api calls and the lock battery.

Reply:
When I call the `if Features.door_sensor` does that cause any polling (I just see the featureValue is pulled in the lockDetails and the passageModeConfig call). How does the polling(or limiting of such) work with the passageModeConfig call?
I'm stuck with this one (stopping polling if None, limiting polling and also order of things [I'm not sure how all the files work together and if they ate concurrent or one after the other) so even a little hint would be helpful with this